docs/arcadia-users-group/20221011-developer-productivity/lesson.md

```diff
@@ -0,0 +1,172 @@
+
+- **Auto Save:** Search for "Auto Save" in the settings menu. And adjust it to the non-default value (which is none). Mine is "afterDelay".
+- **Font Size:** Search for "Font Size" and adjust it as you wish. I like 14pts font.
+- **Trim Trailing Whitespace:**: Search for it and turn it on, so your code doesn't have trailing spaces or tabs.
+- **Insert Final Newline:** Search for it and turn it on. This will prevent you from receiving PR review comments on [this](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline).
+- **Trim Final Newlines:** Again, turn it on. This will make sure there's only a single newline at the end of each file.
+- insertFinalNewline
+- **Format on Save:** Search for "Format on Save" and turn it on. This will make sure, whatever code formatter you use formats the code upon saving it.
+- **Default Terminal application:** Search for "Osx Exex" and type "iTerm.app".
+- **Default Terminal profile:** Search for "Default Profile: Osx" and choose "bash".
+- **Adding code in the PATH:** Open the command palette with Command + Shift + P. Search for "code" and choose the "Install 'code' command in PATH". This will make it so that you can open any file/directory from the terminal with `code`.
+
+### Extensions
+
+Before we go through this section, let's quit VS Code. Renavigate to the correct demo folder and re-open with VS Code:
+
+```{bash}
+cd ~/Desktop/arcadia-computational-training
+code .
+```
+
+For this section, we'll only go through the installation instructions of a few VS Code extensions and give you a list of other extensions you can install on your own time. For downloading and installing extensions, click on the "Extensions" tab from the left tab, search for the name of the extension and hit "install". Here are the extensions:
+
+- Prettier: Customization with "Format Document With"
+- Markdown All in One: Opening Markdown previews with Command + Shift + V
+- Excel Viewer: Opening CSV and Excel previews with Command + Shift + V
+- GitHub Pull Requests and Issues
+- Git Graph
+
+#### Extension recommendations
+
+- Python
+- Pylance - for basic linting
+- R
+- Jupyter - for Jupyter notebooks
+- Jupyter Cell Tags
+- Jupyter Keymap
+- Jupyter Notebook Renderers
+- Jupyter Slide Show
+- One Dark Pro - for color schemes
+- Nextflow specific [extensions](https://github.com/nf-core/vscode-extensionpack) recommended by the nf-core team
+
+## Customizing the terminal
+
+There are 3 main ways, we'll customize our terminals.
+
+- Switching from zsh to bash if you haven't already.
+- Installing [iTerm2](https://iterm2.com/), a better version of Apple's built-in Terminal application.
+- Customizing bash with `.bash_profile`.
+
+### Switching from zsh to bash
+
+Open the Terminal application, run `chsh -s /bin/bash` and quit the application. Next time you open, you should see bash. You'll see an alert that `zsh` the new Apple default when you open the terminal
```

Comment:
I get this is typed up because this is your default, but im wondering what the practical differences are here or if it matters

Reply:
Like the practical differences between bash and zsh or something else? Just so I understand!

Reply:
Ya the practical differences

Reply:
yes I am wondering this as well, with `zsh` being the default on Mac I just switched to using that and therefore all the customizations are in my `.zshrc` file instead of `.bash_profile`. I guess I never took the time to read about the difference between the two and just went with the flow